library/vcenter_gather_moids.py

```diff
@@ -78,6 +80,9 @@ def main():
             object_mo = find_cluster_by_name_datacenter(datacenter_mo, module.params[searched_parameter])
         elif searched_parameter == 'portgroup_name' and module.params[searched_parameter]:
             object_mo = get_mo(content, module.params[searched_parameter], VIM_TYPES[searched_parameter])
+        elif searched_parameter == 'virtualmachine_name' and module.params[searched_parameter]:
+            object_mo = get_mo(content, module.params[searched_parameter], VIM_TYPES[searched_parameter])
+            module.exit_json(changed=False, object_uuid=object_mo.config.instanceUuid)
```

Comment:
You don't need to exit here, just populate object_mo (as you already did) and then when the module exits later it will return the vm uuid using object_id. 
Since all other searches also return object_id, the vm search should not return object_uuid